framework/quota/src/org/apache/cloudstack/quota/QuotaStatementImpl.java

```diff
@@ -0,0 +1,376 @@
+//specific language governing permissions and limitations
+//under the License.
+package org.apache.cloudstack.quota;
+
+import java.math.BigDecimal;
+import java.util.ArrayList;
+import java.util.Calendar;
+import java.util.Date;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import javax.ejb.Local;
+import javax.inject.Inject;
+import javax.naming.ConfigurationException;
+
+import org.apache.cloudstack.framework.config.dao.ConfigurationDao;
+import org.apache.cloudstack.quota.QuotaAlertManagerImpl.DeferredQuotaEmail;
+import org.apache.cloudstack.quota.constant.QuotaConfig;
+import org.apache.cloudstack.quota.dao.QuotaAccountDao;
+import org.apache.cloudstack.quota.dao.QuotaUsageDao;
+import org.apache.cloudstack.quota.vo.QuotaAccountVO;
+import org.apache.log4j.Logger;
+import org.springframework.stereotype.Component;
+
+import com.cloud.user.AccountVO;
+import com.cloud.user.dao.AccountDao;
+import com.cloud.utils.component.ManagerBase;
+
+@Component
+@Local(value = QuotaStatement.class)
+public class QuotaStatementImpl extends ManagerBase implements QuotaStatement {
+    private static final Logger s_logger = Logger.getLogger(QuotaStatementImpl.class);
+
+    @Inject
+    private AccountDao _accountDao;
+    @Inject
+    private QuotaAccountDao _quotaAcc;
+    @Inject
+    private QuotaUsageDao _quotaUsage;
+    @Inject
+    private QuotaAlertManager _quotaAlert;
+    @Inject
+    private ConfigurationDao _configDao;
+
+    final public static int s_LAST_STATEMENT_SENT_DAYS = 6; //ideally should be less than 7 days
+
+    public enum STATEMENT_PERIODS {
+        BIMONTHLY, MONTHLY, QUATERLY, HALFYEARLY, YEARLY
+    };
+
+    private STATEMENT_PERIODS _period = STATEMENT_PERIODS.MONTHLY;
+
+    public QuotaStatementImpl() {
+        super();
+    }
+
+    private void mergeConfigs(Map<String, String> dbParams, Map<String, Object> xmlParams) {
+        for (Map.Entry<String, Object> param : xmlParams.entrySet()) {
+            dbParams.put(param.getKey(), (String)param.getValue());
+        }
+    }
+
+    @Override
+    public boolean configure(String name, Map<String, Object> params) throws ConfigurationException {
+        super.configure(name, params);
+
+        Map<String, String> configs = _configDao.getConfiguration(params);
+
+        if (params != null) {
+            mergeConfigs(configs, params);
+        }
+        String period_str = configs.get(QuotaConfig.QuotaStatementPeriod.key());
+        int period = period_str == null ? 1 : Integer.valueOf(period_str);
+
+        STATEMENT_PERIODS _period = STATEMENT_PERIODS.values()[period];
+        return true;
+    }
+
+    @Override
+    public boolean start() {
+        if (s_logger.isInfoEnabled()) {
+            s_logger.info("Starting Alert Manager");
+        }
+        return true;
+    }
+
+    @Override
+    public boolean stop() {
+        if (s_logger.isInfoEnabled()) {
+            s_logger.info("Stopping Alert Manager");
```

Comment:
This log message seems deceptive.  This class is about quota statements not alerts.